x-pack/plugins/elastic_assistant/server/routes/post_actions_connector_execute.ts

```diff
@@ -240,51 +240,42 @@ export const postActionsConnectorExecuteRoute = (
-          const langChainResponseBody = await callAgentExecutor({
-            alertsIndexPattern: request.body.alertsIndexPattern,
-            allow: request.body.allow,
-            allowReplacement: request.body.allowReplacement,
-            actions,
-            isEnabledKnowledgeBase: request.body.isEnabledKnowledgeBase ?? false,
-            assistantTools,
-            connectorId,
-            elserId,
-            esClient,
-            llmType,
-            kbResource: ESQL_RESOURCE,
-            langChainMessages,
-            logger,
-            onNewReplacements,
-            request,
-            replacements: request.body.replacements,
-            size: request.body.size,
-            telemetry,
-          });
+          const result: StreamFactoryReturnType['responseWithHeaders'] | StaticReturnType =
+            await callAgentExecutor({
+              abortSignal,
+              alertsIndexPattern: request.body.alertsIndexPattern,
+              allow: request.body.allow,
+              allowReplacement: request.body.allowReplacement,
+              actions,
+              isEnabledKnowledgeBase: request.body.isEnabledKnowledgeBase ?? false,
+              assistantTools,
+              connectorId,
+              elserId,
+              esClient,
+              isStream:
+                // TODO implement llmClass for bedrock streaming
+                // tracked here: https://github.com/elastic/security-team/issues/7363
+                request.body.subAction !== 'invokeAI' && actionTypeId === '.gen-ai',
+              llmType: getLlmType(actionTypeId),
+              kbResource: ESQL_RESOURCE,
+              langChainMessages,
+              logger,
+              onNewReplacements,
+              onLlmResponse,
+              request,
+              replacements: request.body.replacements,
+              size: request.body.size,
+              telemetry,
+            });
 
           telemetry.reportEvent(INVOKE_ASSISTANT_SUCCESS_EVENT.eventType, {
             isEnabledKnowledgeBase: request.body.isEnabledKnowledgeBase,
             isEnabledRAGAlerts: request.body.isEnabledRAGAlerts,
           });
 
-          if (conversationId) {
-            // if conversationId is defined, onLlmResponse will be too. the ? is to satisfy TS
-            await onLlmResponse?.(
-              langChainResponseBody.data,
-              langChainResponseBody.trace_data
-                ? {
-                    traceId: langChainResponseBody.trace_data.trace_id,
-                    transactionId: langChainResponseBody.trace_data.transaction_id,
-                  }
-                : {}
-            );
-          }
-          return response.ok({
-            body: {
-              ...langChainResponseBody,
-              replacements: latestReplacements,
-            },
-          });
+          return response.ok<
+            StreamFactoryReturnType['responseWithHeaders']['body'] | StaticReturnType['body']
+          >(result);
```

Comment:
Currently `StaticReturnType[body]` is typed as:
 https://github.com/elastic/kibana/blob/aca83aa6c7ad1fa25493f80d7d0fe47b4d195015/x-pack/plugins/elastic_assistant/server/lib/langchain/types.ts#L13-L21

I think we'll want to use the OAS generated `ExecuteConnectorResponse`, no?

https://github.com/elastic/kibana/blob/46fc9ca0bbdf28f38f35bc16e452d838213e0f10/x-pack/packages/kbn-elastic-assistant-common/impl/schemas/actions_connector/post_actions_connector_execute_route.gen.ts#L48-L68

Reply:
It actually looks like we created the type but did not implement the response body. I will update that type, add the validation for it to the route, and delete my elastic_assistant version. Great catch, thanks @spong!

Reply:
https://github.com/elastic/kibana/pull/174126/commits/d253395440bc411ac1142945aa2ba8d93c66dfed